File: adapter/python/custom_operation.py
# Created by Sezer BOZKIR<admin@sezerbozkir.com> at 16.12.2021
from json_serializer import JsonSerializer


class CustomOperation:
    _json_serializer: JsonSerializer

    def __init__(self, json_serializer: JsonSerializer):
        self._json_serializer = json_serializer

    def serialize_object(self, obj):
        return self._json_serializer.serialize_object(obj)

File: adapter/go/custom-operation.go
package main

type customOperation struct {
	jsonSerializer JSONSerializer
}

func NewCustomOperation(jsonSerializer JSONSerializer) *customOperation {
	return &customOperation{
		jsonSerializer: jsonSerializer,
	}
}

func (c *customOperation) SerializeObject(object []byte) string {
	return c.jsonSerializer.serializeObject(object)
}
